.github/workflows/deploy.yml

```diff
@@ -53,23 +59,39 @@ jobs:
       - name: Build and push docker image from builder target
         uses: docker/build-push-action@v2
         with:
-          build-args: BUILDKIT_INLINE_CACHE=1
+          build-args: |
+            BUILDKIT_INLINE_CACHE=1
+            VERSION=${{ env.TAG }}
           cache-from:
-            ${{ env.DOCKERHUB_REPOSITORY }}:develop
+            ${{ env.DOCKERHUB_REPOSITORY_BASE }}${{ github.event.inputs.environment }}:${{ github.event.inputs.environment }}
           push: true
           tags: |
-            ${{ env.DOCKERHUB_REPOSITORY }}:${{ env.TAG }}
-            ${{ env.DOCKERHUB_REPOSITORY }}:develop
+            ${{ env.DOCKERHUB_REPOSITORY_BASE }}${{ github.event.inputs.environment }}:${{ env.TAG }}
+            ${{ env.DOCKERHUB_REPOSITORY_BASE }}${{ github.event.inputs.environment }}:${{ github.event.inputs.environment }}
           target: production
 
       - name: Deploy to dev
+        if: ${{ github.event.inputs.environment == 'dev' }}
         env:
           AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID }}
           AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
           TF_VAR_paas_user: ${{ secrets.GOVPAAS_DEV_USERNAME }}
           TF_VAR_paas_password: ${{ secrets.GOVPAAS_DEV_PASSWORD }}
         run: |
-          export TF_VAR_paas_app_docker_image=${{ env.DOCKERHUB_REPOSITORY }}:develop
+          export TF_VAR_paas_app_docker_image=${{ env.DOCKERHUB_REPOSITORY_BASE }}${{ github.event.inputs.environment }}:${{ env.TAG }}
           cd terraform/app
           terraform init -reconfigure -input=false -backend-config="bucket=paas-s3-broker-prod-lon-7f2ca242-9929-4662-a79c-c454ea56ea7b"
-          terraform apply -input=false -auto-approve -var-file ../workspace-variables/dev.tfvars -var='secret_paas_app_env_values={"GOVUK_NOTIFY_API_KEY":"${{ secrets.GOVUK_NOTIFY_DEV_API_KEY }}","SECRET_KEY_BASE":"${{secrets.SECRET_DEV_KEY_BASE}}"}'
+          terraform apply -input=false -auto-approve -var-file ../workspace-variables/dev.tfvars -var='secret_paas_app_env_values={"RAILS_MASTER_KEY":"${{secrets.RAILS_MASTER_KEY_DEV}}"}'
+
+      - name: Deploy to staging
+        if: ${{ github.event.inputs.environment == 'staging' }}
+        env:
+          AWS_ACCESS_KEY_ID: ${{ secrets.AWS_ACCESS_KEY_ID_STAGING }}
+          AWS_SECRET_ACCESS_KEY: ${{ secrets.AWS_SECRET_ACCESS_KEY_STAGING }}
+          TF_VAR_paas_user: ${{ secrets.GOVPAAS_STAG_USERNAME }}
+          TF_VAR_paas_password: ${{ secrets.GOVPAAS_STAG_PASSWORD }}
```

Comment:
We can probably do it in a separate PR, but I wonder if we can move any of those to our Rails secrets file?

Reply:
Not as far as I'm aware. I think we can only access those from inside rails